From 0a1c07455e91d13012e6171d2b21b8947ad34480 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Fri, 16 May 2014 11:41:27 -0400
Subject: [PATCH] Do not cache BuddyMenu in favorites view

Let BuddyMenu be created on-demand.

Remove set_registered method, as this palette is now
created on-demand, makes no sense to modify the current
items as these will be set properly in the next
initialization.

Remove _palette_enabled as this is deprecated code
from previous palette implementation.

Remove self._register_menu and simplify code.

This partch avoids #2184 issue for the buddy icon
palette in favorites view.

Signed-off-by: Martin Abente Lahaye <tch@sugarlabs.org>
---
 src/jarabe/desktop/favoritesview.py | 32 ++++++--------------------------
 1 file changed, 6 insertions(+), 26 deletions(-)

diff --git a/src/jarabe/desktop/favoritesview.py b/src/jarabe/desktop/favoritesview.py
index f9ff906..f54b45c 100644
--- a/src/jarabe/desktop/favoritesview.py
+++ b/src/jarabe/desktop/favoritesview.py
@@ -368,7 +368,6 @@ class FavoritesView(ViewContainer):
             alert.props.title = _('Registration Successful')
             alert.props.msg = _('You are now registered '
                                 'with your school server.')
-            self._owner_icon.set_registered()
 
         ok_icon = Icon(icon_name='dialog-ok')
         alert.add_button(Gtk.ResponseType.OK, _('Ok'), ok_icon)
@@ -627,11 +626,6 @@ class OwnerIcon(BuddyIcon):
     def __init__(self, size):
         BuddyIcon.__init__(self, buddy=get_owner_instance(), pixel_size=size)
 
-        self.palette_invoker.cache_palette = True
-
-        self._palette_enabled = False
-        self._register_menu = None
-
         # This is a workaround to skip the callback for
         # enter-notify-event in the parent class the first time.
         def __enter_notify_event_cb(icon, event):
@@ -642,10 +636,6 @@ class OwnerIcon(BuddyIcon):
                                               __enter_notify_event_cb)
 
     def create_palette(self):
-        if not self._palette_enabled:
-            self._palette_enabled = True
-            return
-
         palette = BuddyMenu(get_owner_instance())
 
         client = GConf.Client.get_default()
@@ -653,16 +643,14 @@ class OwnerIcon(BuddyIcon):
             backup_url = client.get_string('/desktop/sugar/backup_url')
 
             if not backup_url:
-                self._register_menu = PaletteMenuItem(_('Register'),
-                                                      'media-record')
+                text = _('Register')
             else:
-                self._register_menu = PaletteMenuItem(_('Register again'),
-                                                      'media-record')
+                text = _('Register again')
 
-            self._register_menu.connect('activate',
-                                        self.__register_activate_cb)
-            palette.menu_box.pack_end(self._register_menu, True, True, 0)
-            self._register_menu.show()
+            register_menu = PaletteMenuItem(text, 'media-record')
+            register_menu.connect('activate', self.__register_activate_cb)
+            palette.menu_box.pack_end(register_menu, True, True, 0)
+            register_menu.show()
 
         self.connect_to_palette_pop_events(palette)
 
@@ -671,14 +659,6 @@ class OwnerIcon(BuddyIcon):
     def __register_activate_cb(self, menuitem):
         self.emit('register-activate')
 
-    def set_registered(self):
-        self.palette.menu_box.remove(self._register_menu)
-        self._register_menu = PaletteMenuItem(_('Register again'),
-                                              'media-record')
-        self._register_menu.connect('activate', self.__register_activate_cb)
-        self.palette.menu_box.pack_end(self._register_menu, True, True, 0)
-        self._register_menu.show()
-
 
 class FavoritesSetting(object):
 
-- 
1.8.3.1

